### PR TITLE
Fix for issue 452

### PR DESCRIPTION
**Pull request by JackMatusiewicz**
There was no spacing applied between the attribute type name and its arguments. This is probably because it was assumed that they'd always be between parentheses. Now Fantomas will check to see if the arg expression for attributes is between parentheses. If it isn't then it will add a space to ensure the code generated still compiles.

### CODE REVIEW

File: src/Fantomas.Tests/AttributeTests.fs [hunk 1]
```diff
@@ -30,6 +30,17 @@ let ``attributes on expressions``() =
 do ()
 """
 
+[<Test>]
+let ``attributes without parentheses on expressions``() =
+    formatSourceString false """
+    [<MyValue 55>]
+    do ()""" config
```
Comment (Smaug123) [inline src/Fantomas.Tests/AttributeTests.fs]: Could you add a test for what happens if there are too many spaces in the attribute? (presumably they get stripped down to 1)

Comment (JackMatusiewicz) [reply src/Fantomas.Tests/AttributeTests.fs]: Yes, will also add similar tests for args in parens

**Comment by jindraivanek** [comment]
Thanks for your contribution! ❤️ 

Nowadays all new changes are done in `trivia` branch (for upcoming Fantomas 3.0). Could you please [retarget](https://github.blog/2016-08-15-change-the-base-branch-of-a-pull-request/) this PR to `trivia` branch?

**Comment by jindraivanek** [comment]
Is this ready?

File: src/Fantomas/Utils.fs [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 open System
 open System.Text.RegularExpressions
+open FSharp.Compiler.Ast
+
+[<RequireQualifiedAccess>]
+module SynExpr =
```
Comment (nojaf) [inline src/Fantomas/Utils.fs]: Does it not make more sense to move this to SourceParser or SourceTransformer?
@jindraivanek this feels inconsistent.

File: src/Fantomas/CodePrinter.fs [hunk 3]
```diff
@@ -377,8 +379,10 @@ and genAttributesCore astContext (ats: SynAttribute seq) =
         match e with
         | ConstExpr(Const "()", _) -> 
             opt sepColonFixed target (!-) -- s
-        | e -> 
-            opt sepColonFixed target (!-) -- s +> genExpr astContext e
+        | e ->
+            let argSpacing =
+                if SynExpr.isInParens e then id else (!- " ")
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs]: Use `sepSpace` instead of `!- " "`

File: src/Fantomas/Utils.fs (hunk 2 above)
Comment (jindraivanek) [reply src/Fantomas/Utils.fs]: Yes, SourceTransformer looks like good place for this.

File: src/Fantomas/CodePrinter.fs (hunk 3 above)
Comment (JackMatusiewicz) [reply src/Fantomas/CodePrinter.fs]: Done

File: src/Fantomas/Utils.fs (hunk 2 above)
Comment (JackMatusiewicz) [reply src/Fantomas/Utils.fs]: Deleted entirely, found SourceTransformer.hasParenthesis that does exactly what I need.

**Pull request merged**